src/functions.pl

```diff
@@ -644,7 +644,10 @@ sub initMapChangeVars {
 	undef %items;
 	undef %spells;
 	undef %incomingParty;
+	
 	undef %talk;
+	AI::clear("NPC");
```

Comment:
TalkNPC task can hook map change event and mark itself as done (`setDone`) in the next `iterate` call.